system/Commands/Utilities/Routes/ControllerFinder.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Commands\Utilities\Routes;
+
+use CodeIgniter\Autoloader\FileLocator;
+use CodeIgniter\Config\Services;
+
+/**
+ * Finds all controllers in a namespace for auto route listing.
+ */
+class ControllerFinder
```

Comment:
Do this class and the other one need to be user extendible? If not, we can make these classes final internal to lessen BC concerns in the future.

Reply:
Thanks! Changed.